crates/mdk-core/src/key_packages.rs

```diff
@@ -2029,4 +2109,333 @@ mod tests {
+        };
+        let base64_mdk = crate::tests::create_test_mdk_with_config(base64_config);
+
+        let (base64_key_package, _) = base64_mdk
+            .create_key_package_for_event(&test_pubkey, relays)
+            .expect("Failed to create base64 key package");
+
+        let hex_size = hex_key_package.len();
+        let base64_size = base64_key_package.len();
+
+        // Base64 should be smaller than hex
+        assert!(
+            base64_size < hex_size,
+            "Base64 ({} bytes) should be smaller than hex ({} bytes)",
+            base64_size,
+            hex_size
+        );
+
+        // Calculate the savings
+        let savings_percent = ((hex_size - base64_size) as f64 / hex_size as f64) * 100.0;
+        println!(
+            "Size comparison: hex={} bytes, base64={} bytes, savings={:.1}%",
+            hex_size, base64_size, savings_percent
+        );
+
+        // Base64 should be approximately 33% smaller (hex is 2x, base64 is 1.33x)
+        // Allow some variance due to encoding overhead
+        assert!(
+            savings_percent > 25.0 && savings_percent < 40.0,
+            "Expected savings between 25-40%, got {:.1}%",
+            savings_percent
+        );
+    }
+
+    #[test]
+    fn test_decode_invalid_hex_string() {
+        let mdk = create_test_mdk();
+
+        // Create a string that has non-hex characters (invalid hex)
+        let invalid = "!!!"; // '!' is not valid for hex
+
+        let result = mdk.decode_key_package_content(invalid, ContentEncoding::Hex);
+
+        // Should fail with a hex decode error
+        assert!(result.is_err());
+        let err_msg = result.unwrap_err().to_string();
+        assert!(
+            err_msg.contains("hex"),
+            "Error should indicate hex format was tried, got: {}",
+            err_msg
+        );
+    }
+
+    #[test]
+    fn test_decode_hex_only_invalid() {
+        let mdk = create_test_mdk();
+
+        // Create a string with only hex characters but odd length (invalid for hex decode)
+        let odd_length_hex = "abc"; // Valid hex chars but odd length
+
+        let result = mdk.decode_key_package_content(odd_length_hex, ContentEncoding::Hex);
+
+        // Should fail with a hex decode error
+        assert!(result.is_err(), "Expected error but got Ok");
+        let err = result.unwrap_err();
+        let err_msg = err.to_string();
+        assert!(
+            err_msg.contains("hex"),
+            "Error should indicate hex format was tried, got: {}",
+            err_msg
+        );
+    }
+
+    #[test]
+    fn test_decode_valid_hex() {
+        let mdk = create_test_mdk();
+
+        // Test that valid hex strings decode successfully
+        let valid_hex = "00000000";
+        let result = mdk.decode_key_package_content(valid_hex, ContentEncoding::Hex);
+        assert!(result.is_ok(), "Valid hex should decode successfully");
+        assert_eq!(result.unwrap(), vec![0, 0, 0, 0]);
+    }
```

Comment:
This test seems a bit redundant given we have the test_key_package_hex_encoding_legacy test